### PR TITLE
Template the project Ballerina.toml file

### LINKED ISSUE #1007: Template the Ballerina.toml file inside the microgateway project
### Description:
Since the Ballerina.toml file created inside the project target directory getting edited due to pilicy hub dependency addition as well as java interceptor platform libraries, we need to template the Ballerina.toml file
### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### LINKED ISSUE #1101: Interceptor module version is ignored during the build
### Description:

When importing a module from [policy hub](central.ballerina.io), module version provided in openapi extension is ignored.
### Steps to reproduce:
1. Add a versioned interceptor module to an openapi file (Ex: `x-wso2-response-interceptor: "praminda/formatters/0.2.0:formatResponse"`)
1. Build the micro-gw project.
1. Inspect the build log and check the version pulled from the policy hub.
   ```
   Compiling source
	   wso2/petstore:3.1.0
	   praminda/formatters:0.3.0 [central.ballerina.io -> home repo]
   ```
1. Latest version of the module is pulled, not the specified version.
### Affected Product Version:

3.1.0-beta
### Environment details (with versions):
- OS: MacOS
- Client:
- Env (Docker/K8s):

### PR DESCRIPTION
**Pull request by praminda**
### Purpose

Templating the `Ballerina.toml` of microgateway project. This will avoid confusions when trying to add/update elements such as dependencies and libs into this file programmatically. 
### Issues

Fixes #1007 
Fixes #1101 
### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

MacOs

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-cli/src/main/resources/templates/ballerinaToml.mustache [hunk 1]
```diff
@@ -1,6 +1,38 @@
...
+    [[platform.libraries]]
+        module = "jms"
+        path = "{{toolkitHome}}/lib/dependencies/org.wso2.securevault-1.1.1.jar"
+        artifactId = "org.wso2.securevault"
+        version = "1.1.1"
+        groupId = "org.wso2.securevault"
+
+    [[platform.libraries]]
+        module = "interceptor"
+        path = "{{toolkitHome}}/lib/dependencies/mgw-interceptor-3.1.0-beta2-SNAPSHOT.jar"
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/resources/templates/ballerinaToml.mustache]: We need to replace the hardcoded version from the pom file

File: components/micro-gateway-cli/src/main/resources/templates/ballerinaToml.mustache [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
 
+[platform]
+target = "java8"
+
+    [[platform.libraries]]
+        module = "jms"
+        path = "{{toolkitHome}}/lib/dependencies/andes-client-3.3.9.jar"
+        artifactId = "andes-client"
+        version = "3.3.9"
+        groupId = "org.wso2.andes.wso2"
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/resources/templates/ballerinaToml.mustache]: artifact id , group id are not supported. Hence shall we remove them

Comment (praminda) [reply components/micro-gateway-cli/src/main/resources/templates/ballerinaToml.mustache]: what are the other unsupported attrs? how about `version`, `module`? Can we remove those too?

Comment (Rajith90) [reply components/micro-gateway-cli/src/main/resources/templates/ballerinaToml.mustache]: I think we can keep only module and path, and remove the rest. Module gives an idea from which module this dependency comes

**Pull request closed without merging**